File: NeetCode/Graph/Find_The_Judge/Find_The_Judge.cpp
// Find_The_Judge.cpp
//
// Compile locally with:
// clang++ Find_The_Judge.cpp -std=c++17 -Wall -Wextra -O2 -o run && ./run
//
// NOTE:
// - Remove main() before submitting to LeetCode.
// - Keep only the class Solution definition.
//

#include <algorithm>
#include <array>
#include <cassert>
#include <cctype>
#include <climits>
#include <cmath>
#include <deque>
#include <functional>
#include <iostream>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <string>
#include <unordered_map>
#include <unordered_set>
#include <utility>
#include <vector>

using namespace std;

class Solution
{
public:
    int findJudge(int n, vector<vector<int>> &trust)
    {
        vector<int> trusts(n, 0);

        for (auto tmp : trust)
        {
            trusts[tmp[0] - 1]--;
            trusts[tmp[1] - 1]++;
        }
        for (int index = 0; index < n; index++)
        {
            if (trusts[index] == n - 1)
            {
                return index + 1;
            }
        }
        return -1;
    }
};

int main()
{
    cout << "Running Find_The_Judge.cpp..." << endl;

    Solution sol;

    // TODO:
    // Add local test calls here

    return 0;
}

File: NeetCode/Graph/Find_The_Judge/find__the__judge.py
from typing import Optional, List
import heapq
from collections import defaultdict


class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        trusted = [set() for _ in range(n)]
        trusts = [set() for _ in range(n)]

        for a, b in trust:
            trusted[b - 1].add(a - 1)
            trusts[a - 1].add(b - 1)

        for index in range(n):
            if len(trusted[index]) == n - 1 and len(trusts[index]) == 0:
                return index + 1
        return -1

        pass


if __name__ == "__main__":
    sol = Solution()
    n = 1
    trust = []
    print(sol.findJudge(n, trust))
    print("Running Solution...")
